Dockerfile

```diff
@@ -0,0 +1,17 @@
+# installing node
+FROM node:18.6.0 as builder
+RUN mkdir -p /code
+WORKDIR /code
+ADD package* /code
+
+ENV NODE_OPTIONS=--openssl-legacy-provider
+
+COPY yarn.lock .
+RUN yarn install --ignore-engines
+
+COPY contracts contracts
+COPY scripts scripts
+COPY lib lib
+COPY hardhat.config.js .
+
+RUN npx hardhat compile
```

Comment:
We need a CMD so there is an entry point.
@mohandast52 
You have this in your working dockerfile.
Please append it here.